java/code/src/com/redhat/rhn/manager/channel/repo/BaseRepoCommand.java

```diff
@@ -216,11 +220,15 @@ public void store() throws InvalidRepoUrlException, InvalidRepoLabelException,
         }
 
         if (this.url != null && this.type != null) {
+            UrlValidator urlValidator = new UrlValidator();
             ContentSourceType cst = ChannelFactory.lookupContentSourceType(this.type);
             boolean alreadyExists = !ChannelFactory.lookupContentSourceByOrgAndRepo(
                     org, cst, url).isEmpty();
```

Comment:
Let's move these two lines after the `isValid` check too, since this is the expensive part where a DB query happens.

Reply:
Right, my previous change didn't actually do anything in that regard.